src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -3353,52 +3535,20 @@ private static string GetTimeZoneDirectory()
         // BSD                              July 18, 2003                             BSD
         // 
         // 
-
-        //
-        // TZif_CalculateTransitionTime -
-        //
-        // Example inputs:
-        // ----------------- 
-        // utc               =     1918-03-31T10:00:00.0000000Z
-        // transitionType    =     {-08:00:00  DST=False,  Index 4}
-        // standardTime      =     False
-        // gmtTime           =     False
-        //
-        static private TransitionTime TZif_CalculateTransitionTime(DateTime utc, TimeSpan offset,
-                                                                   TZifType transitionType, Boolean standardTime,
-                                                                   Boolean gmtTime, out DateTime ruleDate) {
-
-            // convert from UTC to local clock time
-            Int64 ticks = utc.Ticks + offset.Ticks;
-            if (ticks > DateTime.MaxValue.Ticks) {
-                utc = DateTime.MaxValue;
-            }
-            else if (ticks < DateTime.MinValue.Ticks) {
-                utc = DateTime.MinValue;
-            }
-            else {
-                utc = new DateTime(ticks);
-            }
-
-            DateTime timeOfDay = new DateTime(1, 1, 1, utc.Hour, utc.Minute, utc.Second, utc.Millisecond);
-            int month = utc.Month;
-            int day = utc.Day;
-
-            ruleDate = new DateTime(utc.Year, month, day);
-            // FUTURE: take standardTime/gmtTime into account
-            return TransitionTime.CreateFixedDateRule(timeOfDay, month, day);
-        }
-
-        static private void TZif_GenerateAdjustmentRules(out AdjustmentRule[] rules, DateTime[] dts, Byte[] typeOfLocalTime,
-                                                         TZifType[] transitionType, Boolean[] StandardTime, Boolean[] GmtTime) {
+        static private void TZif_GenerateAdjustmentRules(out AdjustmentRule[] rules, TimeSpan baseUtcOffset, DateTime[] dts, Byte[] typeOfLocalTime,
+            TZifType[] transitionType, Boolean[] StandardTime, Boolean[] GmtTime, string futureTransitionsPosixFormat)
+        {
             rules = null;
 
             int index = 0;
             List<AdjustmentRule> rulesList = new List<AdjustmentRule>(1);
-            bool succeeded = true;
-           
-            while (succeeded && index < dts.Length) {
-                succeeded = TZif_GenerateAdjustmentRule(ref index, ref rulesList, dts, typeOfLocalTime, transitionType, StandardTime, GmtTime);
+
+            if (dts.Length > 0)
```

Comment:
I wouldn't say it is 'common', but it does occur for a small number of time zones, so I will make the suggested optimization.